### DISCUSSION TITLE
Wiki software with MyST support?

### DISCUSSION THREAD

**Discussion by tavin** [executablebooks/meta#978] [General]
Does anyone know of such a thing? I've been trying to use Jupyter for live markdown editing (with preview) but it's not quite "there" yet in several ways. I need latex math support and myst-like markdown. I don't need it hosted on a server with a database.

Googling this idea tends to lead to wikis about an old game some of you may remember :)

**Comment by sneakers-the-rat** [comment]
this is EXACTLY WHAT I HAVE BEEN WONDERING THIS WEEK and I simply must ask @rowanc1 what it would take to make curvenote into a tiddlywiki-like wiki

edit: OR a more powerful wiki, I have been talking to some of the fedwiki ppl and markdown notation has been raised!!!!

**Reply by rowanc1** [reply]
**Ambitious Answer**
One of the main goals that I see that can be enabled with MyST (especially the structured data parts) are to work in the scholarly ecosystem to help turn open-science resources into something that more resembles a wiki with rich cross-linking aspects of collaboration, and a merging between public and private work. Maybe that is some version of a fediwiki, I am not really sure about the terminology, however, what it needs to support is a rich spectrum between tools for close team collaboration (maybe private) that can eventually evolve and be linked into the wider ecosystem. Wikis (as they are traditionally used) have slightly different goals for creating works -- the goal there is often to get to consensus around a topic/article, whereas research articles/notes/lectures/knowledge is a lot more personal and has a different cadence to being updated. There is a very rich space to explore on that spectrum, and I certainly hope one that MyST / ExecutableBooks can help facilitate!

@agoose77 recently asked me why I care about the MyST community, I gave him a long answer which I think lays out some of the key pieces that I see MyST/ExecutableBooks solving in this space (📺, 10min on 2x):
https://www.loom.com/share/28898917ad054d38b7e868ca6b73336c

The last half is kinda relevant here for some of the goals around creating linked knowledge, reusing content, and networked, so thought I would share. :)

**Still Ambitious, but a more direct answer**
@sneakers-the-rat please help bring MyST into those fediwiki conversations, our community shares so many of the same goals, and is tackling them from a different angle, it would be so cool to see them connect.

@tavin I think you are aware of `jupyterlab-myst`, it is getting good, and should hopefully support previews of markdown files soon, not just notebooks as it does now. I think that could be a cool way to bootstrap some of these wiki-like environments that also have access to computation.

**Reply by yuvipanda** [reply]
A long time ago, in a far away land, I added support for Jupyter notebook rendering to Mediawiki (the software behind Wikipedia) - https://www.mediawiki.org/wiki/Extension:NotebookViewer (code in https://github.com/wikimedia/mediawiki-extensions-NotebookViewer). Something like that can also be done for MyST, if anyone wants to.

**Reply by rowanc1** [reply]
Oh, that is so cool! Excited to check that out. 🚀

**Reply by tavin** [reply]
@rowanc1 if we can get markdown previews working with jupyterlab-myst, and there is at least some avenue for extensibility (like the custom directives and syntax toggles that become possible with my poc fork), then I think we're nearly there.

All that I really need on top of that is a toolbar that opens the visible markdown preview in the file tree or directly in the markdown editor. And for this to work on files that don't exist (yet -- arrived at by following a link).

The sum of these features amounts to a live jupyter book and when I'm satisfied with my edits, I can build and push to github. If a friend or colleague has edits they can run the same lab environment and send me a PR. A p2p/offline wiki of sorts... which a more ambitious person can turn into a wiki server (take jupyterlab plus key extensions and strip it down?).

**Reply by sneakers-the-rat** [reply]
exactly :) a tiddlywiki made of myst <3 https://tiddlywiki.com/

**Reply by tavin** [reply]
So fwiw I looked into integrating myst and tiddlywiki's [markdown](https://github.com/Jermolene/TiddlyWiki5/tree/master/plugins/tiddlywiki/markdown) plugin, which is based on markdown-it and compatible with markdown-it extensions. Superficially it's not very hard to pull in [markdown-it-docutils](https://github.com/executablebooks/markdown-it-docutils) or [markdown-it-myst](https://github.com/executablebooks/mystjs/tree/main/packages/markdown-it-myst). However,

* markdown-it-docutils appears to be dead, in favor of markdown-it-myst
* markdown-it-myst afaict will parse but not render directives and roles

I haven't yet figured out if there's another library that will do the rendering, or if it's a todo for markdown-it-myst, or if users of markdown-it-myst are intended to write a renderer. Hopefully someone on this thread knows?

I guess it's also possible to fork tiddlywiki's markdown plugin and replace all the markdown-it stuff with "native" mystjs but I can't say how much effort it would take.

Feeling kinda confused about which direction to take here, friends! It could be a long time before the desired jupyterlab-myst features land? Do markdown-it and markdown-it-myst have a future?
